packages/core/src/tools/sanitize.spec.ts

```diff
@@ -0,0 +1,193 @@
+    })
+
+    it('should handle functions', () => {
+      function testFunction() {
+        return true
+      }
+      if (isIE()) {
+        // IE does not provide access to function name
+        expect(sanitize(testFunction)).toBe('[Function] unknown')
+      } else {
+        expect(sanitize(testFunction)).toBe('[Function] testFunction')
+      }
+    })
+
+    it('should handle bigint', () => {
+      const bigIntFunction: (val: number) => any = (window as any).BigInt
+      if (typeof bigIntFunction === 'function') {
+        const bigint = bigIntFunction(2)
+        expect(sanitize(bigint)).toEqual('[BigInt] 2')
+      } else {
+        pending('BigInt is not supported on this browser')
+      }
+    })
+  })
+
+  describe('objects handling', () => {
+    it('should serialize a Date as a string', () => {
+      const date = new Date('2022-12-12')
+      expect(sanitize(date)).toBe('2022-12-12T00:00:00.000Z')
+    })
+
+    it('should not traverse instrumented DOM nodes', () => {
+      const node = document.createElement('div')
+      ;(node as any).__hiddenProp = { value: 42 }
+
+      expect(sanitize(node)).toBe('[HTMLDivElement]')
+    })
+
+    it('should serialize events', () => {
+      let event: CustomEvent
+      if (isIE()) {
+        event = document.createEvent('CustomEvent')
+        event.initCustomEvent('MyEvent', false, false, {})
+      } else {
+        event = new CustomEvent('MyEvent')
+      }
+
+      expect(sanitize(event)).toEqual({
+        type: 'MyEvent',
+        isTrusted: false,
+      })
+    })
+
+    it('should serialize errors', () => {
+      const error = new SyntaxError('Wrong syntax')
+      expect(sanitize(error)).toEqual({
+        message: 'Wrong syntax',
+        type: 'SyntaxError',
+        stack: jasmine.any(String),
+      })
+    })
+
+    it('should serialize objects like maps as a string', () => {
+      const map = new Map([
+        ['a', 13],
+        ['b', 37],
+      ])
+      if (isIE()) {
+        // IE does not distinguish maps, weakmaps, sets... from generic objects
+        expect(sanitize(map)).toEqual({})
+      } else {
+        expect(sanitize(map)).toBe('[Map]')
+      }
+    })
+  })
+
+  describe('arrays handling', () => {
+    // JSON.stringify ignores properties on arrays - We replicate the behavior
+    it('should ignore non-numerical properties on arrays', () => {
+      const arr = [1, 2, 3, 4]
+      ;(arr as any)['test'] = 'test'
+
+      expect(sanitize(arr)).toEqual([1, 2, 3, 4])
+    })
+  })
+
+  describe('circular references handling', () => {
+    it('should remove circular references', () => {
+      const obj: any = { a: 42 }
+      obj.self = obj
+
+      expect(sanitize(obj)).toEqual({ a: 42, self: '[Circular Ref]' })
+    })
+
+    it('should replace already visited objects with a json path', () => {
+      const inner = [1]
+      const obj = { a: inner, b: inner }
+
+      expect(sanitize(obj)).toEqual({ a: [1], b: '[Visited] $.a' })
+    })
```

Comment:
The path for the first element of an array `a` inside the root object would read `$.a.0`, instead of `$.a[0]`. This is readable but will not work with all parsers.

Reply:
Got it, IMO it could be easier to read with `[]` notation but I'm ok with it if it adds too much extra complexity.
Could we at least have a test for the current behaviour?

Reply:
Keeping simple implementation for now - Added test